### PR TITLE
TRA-3420 inform user of tls traffic

### PR DESCRIPTION
**Pull request by RamiBerm**

### CODE REVIEW

File: tap/passive_tapper.go [hunk 1]
```diff
@@ -429,6 +429,21 @@ func startPassiveTapper(harWriter *HarWriter, outboundLinkWriter *OutboundLinkWr
...
 			assembler.AssembleWithContext(packet.NetworkLayer().NetworkFlow(), tcp, &c)
 			assemblerMutex.Unlock()
+
+			sourceIP := packet.NetworkLayer().NetworkFlow().Src().String()
+			destinationIP := packet.NetworkLayer().NetworkFlow().Dst().String()
+			destinationPort := int(tcp.DstPort)
+
+			if streamFactory.getStreamProps(sourceIP, destinationIP, destinationPort).isTapTarget {
+				fmt.Printf("%s -> %s:%d isTapTarget:%t", sourceIP, destinationIP, destinationPort, streamFactory.getStreamProps(sourceIP, destinationIP, destinationPort).isTapTarget)
+				if IsTLSHandshakePacket(tcp.Payload) {
```
Comment (nimrod-up9) [inline tap/passive_tapper.go:439]: Won't there only be a handshake when the there's a new stream?
Why do we need to run this for every single tcp packet?
Also, is it possible for the handshake to be split between several tcp packets? In that case we would need to look for the handshake after tcp assembly.

File: ui/src/App.tsx [hunk 2]
```diff
@@ -15,11 +17,40 @@ const useStyles = makeStyles(() => ({
...
+    const [addressesWithTLS, setAddressesWithTLS] = useState(new Set());
+
+    useEffect(() => {
+        (async () => {
+            const recentTLSLinks = await api.getRecentTLSLinks();
+
+            if (recentTLSLinks?.length > 0) {
+                for (const tlsDestAddress of recentTLSLinks) {
+                    addressesWithTLS.add(tlsDestAddress)
+                }
```
Comment (nimrod-up9) [inline ui/src/App.tsx:38]:
```typescript
addressesWithTLS = [...addressesWithTLS, ...tlsDestAddress]
```

File: ui/src/App.tsx [hunk 3]
```diff
@@ -15,11 +17,40 @@ const useStyles = makeStyles(() => ({
...
+                setAddressesWithTLS(new Set(addressesWithTLS));
+                setShowTLSWarning(true);
+            }
+
+        })();
+    }, []);
+
+    const onTLSDetected = (destAddress: string) => {
+        addressesWithTLS.add(destAddress);
+        setAddressesWithTLS(new Set(addressesWithTLS));
```
Comment (nimrod-up9) [inline ui/src/App.tsx:48]: Why do we need two ways to `setAddressesWithTLS`?

File: tap/passive_tapper.go [hunk 4]
```diff
@@ -429,6 +429,21 @@ func startPassiveTapper(harWriter *HarWriter, outboundLinkWriter *OutboundLinkWr
 			assemblerMutex.Lock()
 			assembler.AssembleWithContext(packet.NetworkLayer().NetworkFlow(), tcp, &c)
 			assemblerMutex.Unlock()
+
+			sourceIP := packet.NetworkLayer().NetworkFlow().Src().String()
+			destinationIP := packet.NetworkLayer().NetworkFlow().Dst().String()
+			destinationPort := int(tcp.DstPort)
+
+			if streamFactory.getStreamProps(sourceIP, destinationIP, destinationPort).isTapTarget {
+				fmt.Printf("%s -> %s:%d isTapTarget:%t", sourceIP, destinationIP, destinationPort, streamFactory.getStreamProps(sourceIP, destinationIP, destinationPort).isTapTarget)
```
[the remark this answers is not in the document]
Comment (RamiBerm) [reply tap/passive_tapper.go:438]: whoops

File: ui/src/App.tsx (hunk 3 above)
Comment (RamiBerm) [reply ui/src/App.tsx:48]: This is for notifications from websocket, the previous one is the result of the http request done on page load

**Review by nimrod-up9 — APPROVED** [review]

**Pull request merged**